### PR TITLE
[Paywall Experiment] Reviews variant

### LINKED ISSUE #2031: Social Variation: Add the Rating Views

### PR DESCRIPTION
**Pull request by danielebogo**
| 📘 Part of: #1986
|:---:|

Fixes #2031

This PR introduces the last Paywall variant: Reviews. We want to display a list of Reviews.

https://github.com/user-attachments/assets/c333a026-1e5f-4758-9f0e-baed34aa7f25

## To test

> [!NOTE]
> Make sure to enable the upgradeExperiment feature flag

- CI must be green 🟢 
- Use a non-plus account
- Go to `Profile -> Settings` and tap on `Pocket Casts Plus`
- Check the new paywall layout like the video above

## Checklist

- [ ] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Modules/Server/Sources/PocketCastsServer/Public/Sharing/Structs/ServerConstants.swift [hunk 1]
```diff
@@ -46,6 +46,7 @@ public enum ServerConstants {
         public static let pocketcastsDotCom = "https://pocketcasts.com/"
         public static let automatticDotCom = "https://automattic.com/"
         public static let automatticWorkWithUs = "https://automattic.com/work-with-us/"
+        public static let appStore = "https://apps.apple.com/app/id414834813"
```
Comment (danielebogo) [inline Modules/Server/Sources/PocketCastsServer/Public/Sharing/Structs/ServerConstants.swift:49]: I may be wrong but I can't find a way to open the App Store on the Reviews page. @leandroalonso @bjtitus any ideas?

Comment (leandroalonso) [reply Modules/Server/Sources/PocketCastsServer/Public/Sharing/Structs/ServerConstants.swift:49]: I don't think it's possible. All you can do is open in the app's page.

**Review by leandroalonso — APPROVED** [review]
✅ Tested the mentioned steps and it works as expected
✅ Tested with the flag disables
✅ Tested with slumber enabled/disabled
✅ Tested the event when tapping "Get Pocket Casts Plus"

**Pull request merged**